lib/cocoapods/downloader/cache.rb

```diff
@@ -282,11 +282,12 @@ def in_tmpdir(&blk)
       def copy_and_clean(source, destination, spec)
         specs_by_platform = group_subspecs_by_platform(spec)
         destination.parent.mkpath
+        Cache.write_lock(source) do
+          Sandbox::PodDirCleaner.new(source, specs_by_platform).clean!
```

Comment:
This is the change. 

The `source` folder here is a temp folder. If many podspecs are supplied by a given repo we end up copying the entire repo many many times and then globbing everything for each podspec we want in order to clean up the final directory for the pod.

This does not make sense and instead we now clean up the source repo first across all podspecs. This means we glob once and then copy over the pristine files we wish to store in the cache.